Makefile

```diff
@@ -20,5 +20,5 @@ clean:
 ci:
 	pre-commit run --all-files --show-diff-on-failure && \
 	export PYTHONPATH=$(MKFILE_DIR) && \
-	cuda-memcheck pytest $(PYTEST_CI_ARGS) tests/ && \
+	cuda-memcheck --leak-check full pytest $(PYTEST_CI_ARGS) tests/ && \
```

Comment:
Should we add `--error-exitcode 1`? Right now this will report memory leaks, which seems to be in one of the tests already, but doesn't trigger a failure

Reply:
Good call, I didn't realize this was necessary. Added in c03150e.

Also interesting that we get a [failure](https://jenkins.relaytxinternal.com/job/timemachine/job/PR-574/1/display/redirect) even without this. This seems like an independent issue possibly exposed by adding the `cudaDeviceReset()` calls?

Reply:
> Also interesting that we get a failure even without this

This looks to be resolved by 5cc6dd5. In an offline debugging session, we figured out that `cudaDeviceReset()` was running before GC (and hence destructors) in some cases. Adding an explicit `gc.collect()` before calling `custom_ops.cuda_device_reset()` seems to be a fix.